src/app/page/Signup/SignupScene.jsx

```diff
@@ -46,59 +45,65 @@ function SignupScene(props) {
     };
   }
 
-  function handleSignupSuccess(user) {
-    setValues({
-      ...values,
+  async function handleSignupSuccess(user) {
+    let data = {
+      ...getPayload(),
       phoneNumber: user.phoneNumber,
       displayName: user.displayName,
       email: user.email,
       photoURL: user.photoURL,
-    });
-    updateUser();
-    setActiveTab(Tabs.CONNECT);
+      uid: user.uid,
+    };
+    await updateUser(data, () => setActiveTab(Tabs.CONNECT));
     return false;
   }
-  function handleConnectSuccess(user) {
-    setValues({
-      ...values,
+
+  async function handleConnectSuccess(user) {
+    let data = {
+      ...getPayload(),
       displayName: user.displayName,
       email: user.email,
       photoURL: user.photoURL,
-    });
-    updateUser();
-    setActiveTab(Tabs.PROFILE);
+      uid: user.uid,
+    };
+    await updateUser(data, () => setActiveTab(Tabs.PROFILE));
     return false;
   }
-  async function updateUser() {
-    const payload = getPayload();
 
+  async function handleProfileSuccess() {
+    await updateUser({ ...getPayload() }, () => setActiveTab(Tabs.SUCCESS));
+  }
+
+  async function updateUser(payload, callback) {
+    payload = getPayloadWithAddress(payload);
+    try {
+      User.update(payload.uid, payload).then(() => {
+        setValues({ ...payload });
+        callback();
+      });
+    } catch (error) {
+      setErrorSnackbarMessage(error);
+      setValues({ ...payload });
+      callback();
+    }
+  }
+
+  async function getPayloadWithAddress(payload) {
     let location = payload?.location;
     try {
       if (location.address) {
         const data = await addressLookup(location.address);
         location = { ...location, lat: data.lat, lng: data.long };
+        payload = { ...payload, location: location };
       }
     } catch (error) {
-      console.log("ERROR WHEN GETTiNG LOCATION", error);
+      console.log("ERROR WHEN GETTING LOCATION", error);
       console.log("address: ", location.address);
```

Comment:
remove this

Reply:
I removed it and set the error snackbar message. We don't expect it to occur, but this way we'll see something if it does. (I also added an await in the updateUser)